### PR TITLE
Activity name validation

### LINKED ISSUE #13: Activity name validation
In #4 we have pending test with which we want to ensure that our model will not be saved in DB if no name is given or such name already exists in DB. 

We need to express this desire in rspec model test.

help:

https://relishapp.com/rspec/rspec-rails/v/3-1/docs/model-specs
http://guides.rubyonrails.org/active_record_validations.html

hints

* populate model via <modelname>.new without parameters and check if that object is `valid?`
* use rspec syntax with `expect` like `expect(object.valid).to be true`
* to accomplish the task we need to have a validation in the Activity model (see above materials to read)
* here we need at least two validations one is for present name and one is for it's uniqueness

### PR DESCRIPTION
**Pull request by thutterer**
closes #13

### CODE REVIEW

File: app/models/activity.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class Activity < ActiveRecord::Base
   has_and_belongs_to_many :users
   has_one :creator, class_name: User
+  validates_presence_of :name
```
Comment (kalabiyau) [inline app/models/activity.rb]: @thutterer  it is really better to use new format of validations
like 

validates <column_name_symbol>, presence: true

File: spec/models/activity_spec.rb [hunk 2]
```diff
@@ -1,9 +1,12 @@
 require 'rails_helper'
 
 describe Activity do
+
   it { should have_and_belong_to_many(:users) }
   
-  it 'is not being saved without a name'
-  it 'will not be saved if there is a activity with the same name '
+  subject { create(:activity) }
```
Comment (kalabiyau) [inline spec/models/activity_spec.rb]: we do not actually need a created instance here. It should work just fine even without that. As subject by default is always a new instance of described_class which is in this case `Activity`

**Comment by kalabiyau** [comment]
Please fix those two things and you can merge

File: spec/models/activity_spec.rb (hunk 2 above)
Comment (thutterer) [inline spec/models/activity_spec.rb]: i thought so too at first. but without that line, I get this when running rspec:
```ruby
Failures:

  1) Activity should require case sensitive unique value for name
     Failure/Error: it { should validate_uniqueness_of(:name) }
     NoMethodError:
       undefined method `primary' for #<ActiveRecord::ConnectionAdapters::Column:0x00000008c43bc0>
     # ./spec/models/activity_spec.rb:8:in `block (2 levels) in <top (required)>'
     # ./spec/rails_helper.rb:59:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:58:in `block (2 levels) in <top (required)>'
```

Comment (kalabiyau) [inline spec/models/activity_spec.rb]: Ok, leave it for now, I will look into this.

**Pull request merged**